### PR TITLE
Adding integration matrix integration tests to real HA instance

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

Adding integration tests with real Home Assistant instance

Using matrix feature of GH Action we integration test to latest version of HA, dev release and current beta in combination with or without the latest version of NetDaemon integration.

The first MVP is simple, just a command line exe that are run from the action. I like to start simple and grow from there. 

To do in other pr: 
- [x] Add more test cases
- [x] Refactor to use test project instead of command line

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: .github/workflows/ci_integration_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - main
+      - dev
+
+jobs:
+  test_with_home_assistant:
+    name: Integration tests with Home Assistant
+    strategy:
+      matrix:
+        channel: [stable, beta, dev]
+        with_interation: ["yes", "no"]
```
Comment (ludeeus) [inline .github/workflows/ci_integration_tests.yaml:19]:
```suggestion
        with_interation: [yes, no]
```
Maybe without quotes :shrug:  (it's currently not working)

Comment (ludeeus) [inline .github/workflows/ci_integration_tests.yaml:19]:
```suggestion
        with_integration: ["yes", "no"]
```

File: .github/workflows/ci_integration_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+name: Test with Home Assistant
+
+on:
```
Comment (ludeeus) [inline .github/workflows/ci_integration_tests.yaml:3]: Should trigger on schedule as well

File: .github/workflows/ci_integration_tests.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        with_integration: [yes, no]
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📥 Checkout the repository
+        uses: actions/checkout@v2
+
+      - name: 🥅 Install .Net 5
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '5.0.x' # SDK Version
```
Comment (ludeeus) [inline .github/workflows/ci_integration_tests.yaml:28]: Maybe change this to global var and put it at the top of the file so you don't have to "scan" for it when you are updating

File: .github/workflows/ci_integration_tests.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        uses: ludeeus/setup-homeassistant@main
+        with:
+          tag: ${{ matrix.channel }}
+          config-dir: ./tests/Integration/HA/config
+#
+
+      - name: 🧪 Run integration tests
+        env:
+          HOMEASSISTANT__TOKEN: ${{ steps.homeassistant.outputs.token }}
+          HOMEASSISTANT__HOST: "127.0.0.1"
```
Comment (ludeeus) [inline .github/workflows/ci_integration_tests.yaml:54]: I should really make this available as ${{ steps.homeassistant.outputs.host }} in my action :/

File: tests/Integration/HA/config/configuration.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
+# Configure a default setup of Home Assistant (frontend, api, etc)
```
Comment (ludeeus) [inline tests/Integration/HA/config/configuration.yaml:2]: This should just stubbed and appended to the default file that Home Assistatn creates

File: tests/Integration/HA/config/secrets.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
+# Use this file to store secrets like usernames and passwords.
+# Learn more at https://www.home-assistant.io/docs/configuration/secrets/
```
Comment (ludeeus) [inline tests/Integration/HA/config/secrets.yaml:3]: With the prevoous comment all these dummy files are not needed

File: tests/Integration/HA/config/configuration.yaml (hunk 5 above)
Comment (ludeeus) [reply tests/Integration/HA/config/configuration.yaml:2]: Disregard, this is not possible without changing the action

File: tests/Integration/HA/config/secrets.yaml (hunk 6 above)
Comment (ludeeus) [reply tests/Integration/HA/config/secrets.yaml:3]: Disregard, this is not possible without changing the action

**Review by ludeeus — APPROVED** [review]
:heavy_check_mark: Action parts look good, remaining comments can be addressed later when I have updated `setup-homeassistant` so we can implement my "resolved" comments as well.

**Review by helto4real — COMMENTED** [review]
>  Action parts look good, remaining comments can be addressed later when I have updated `setup-homeassistant` so we can implement my "resolved" comments as well.

Thanks @ludeeus yea I plan to do another PR really soon but I am eager to get this live as soon as possible cause it gives great value.

File: .github/workflows/ci_integration_tests.yaml (hunk 3 above)
Comment (helto4real) [reply .github/workflows/ci_integration_tests.yaml:29]: Good idea, another PR

**Pull request merged**